src/main/java/cz/it4i/parallel/ImageJServerWorker.java

```diff
@@ -105,95 +104,112 @@ public Dataset importData(Path path) {
 
 	@Override
 	public void exportData(Dataset dataset, Path p) {
-		String filePath = p.toString();
-		String id = mockedData2id.get(dataset);
-		String getUrl = "http://" + hostName + ":" + String.valueOf(port) + "/objects/" + id + "/" + getImageType(filePath);
-		HttpClient httpClient = HttpClientBuilder.create().build();
-		HttpGet get = new HttpGet(getUrl);
+
+		final String filePath = p.toString();
+		final String objectId = mockedData2id.get(dataset);
 
 		try {
 
-			HttpResponse response = httpClient.execute(get);
-			HttpEntity entity = response.getEntity();
+			String getUrl = "http://" + hostName + ":" + String.valueOf(port) + "/objects/" + objectId + "/"
+					+ getImageType(filePath);
+			HttpGet httpGet = new HttpGet(getUrl);
+
+			HttpEntity entity = HttpClientBuilder.create().build().execute(httpGet).getEntity();
+
 			if (entity != null) {
-				// long len = entity.getContentLength();
-				BufferedInputStream bis = new BufferedInputStream(entity.getContent());
-				BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(new File(filePath)));
-				int inByte;
-				while ((inByte = bis.read()) != -1) {
-					bos.write(inByte);
-				}
-				bis.close();
-				bos.close();
 
-			}
+				BufferedInputStream bis = null;
+				BufferedOutputStream bos = null;
+
+				try {
+
+					bis = new BufferedInputStream(entity.getContent());
```

Comment:
It could be placed into "try" section for autoclosing

Reply:
Done.